File: 0456-132-pattern/0456-132-pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {
        stack<int> s;
        int third = INT_MIN;
        for(int i=nums.size()-1; i>=0; i--){
            if(third > nums[i]) return true;
            while(!s.empty() && nums[i] > s.top()){
                third = s.top();
                s.pop();
            }
            s.push(nums[i]);
        }
        return false;
    }
};

File: 0456-132-pattern/0456-132-pattern.py
class Solution {
    public boolean find132pattern(int[] nums) {
        Stack<Integer> s = new Stack<>();
        int third  = Integer.MIN_VALUE;
        for(int i=nums.length-1; i>=0; i--){
            if(third > nums[i]) return true;
            while(!s.isEmpty() && s.peek() < nums[i]){
                third = s.pop();
            }
            s.push(nums[i]);
        }
        return false;
    }
}